sdk/src/event_listener/mod.rs

```diff
@@ -0,0 +1,100 @@
+use crate::events_watcher::evm::{Config as ConfigT, EventWatcher};
+use crate::events_watcher::substrate::SubstrateEventWatcher;
+use alloy_network::Ethereum;
+use async_trait::async_trait;
+
+pub mod periodic;
+
+/// The [`EventListener`] trait defines the interface for event listeners.
+#[async_trait]
+pub trait EventListener<T: Send + Sync + 'static>: Send + Sync + 'static {
+    /// Obtains the next event to be processed by the event listener.
+    async fn next_event(&mut self) -> Option<T>;
+
+    /// Logic for handling received events
+    async fn handle_event(&mut self, event: T) -> std::io::Result<()>;
+
+    /// The event loop for the event listener
+    async fn execute(&mut self) {
+        while let Some(event) = self.next_event().await {
+            if let Err(err) = self.handle_event(event).await {
+                crate::error!("Error handling event: {err}");
+            }
+        }
+
+        crate::warn!("Event listener has stopped")
+    }
+}
+
+pub struct EthereumWatcherWrapper<T: EventWatcher<C>, C: ConfigT<N = Ethereum>> {
+    listener: T,
+    _phantom: std::marker::PhantomData<C>,
+}
+
+pub struct SubstrateWatcherWrapper<
+    T: SubstrateEventWatcher<C>,
+    C: subxt::Config + Send + Sync + 'static,
+> {
+    listener: T,
+    _phantom: std::marker::PhantomData<C>,
+}
+
+impl<Config: ConfigT<N = Ethereum>, Watcher: EventWatcher<Config>> From<Watcher>
+    for EthereumWatcherWrapper<Watcher, Config>
+{
+    fn from(value: Watcher) -> Self {
+        Self {
+            listener: value,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+
+// TODO: Refactor up a level of abstraction after tests pass to truly use the EventListener interface
```

Comment:
I made a mistake calling it Eigenlayer protocol because I don't want you to associate EVM listeners with only Eigenlayer! They can be used on Tangle too. So we should just require you to specify an event listener and that will be the trigger for job execution. EVM listeners can also be instantiated in the `main.rs` to listen to contracts on any blockchain by setting up providers and contracts against any RPC anywhere. Let me know if this makes sense!